src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java

```diff
@@ -208,19 +215,29 @@ public synchronized void connect() {
                 clientIdentification.getClientId(),
                 permits,
                 permitsBatch,
-                e -> scheduleReconnect(),
+                this::onConnectionError,
                 this::registerOutboundStream
         );
 
         //noinspection ResultOfMethodCallIgnored
         queryServiceStub.openStream(responseObserver);
-        StreamObserver<QueryProviderOutbound> newValue = responseObserver.getInstructionsForPlatform();
+        CallStreamObserver<QueryProviderOutbound> newValue = responseObserver.getInstructionsForPlatform();
 
-        supportedQueries.forEach(k -> newValue.onNext(
+        supportedQueries.keySet().forEach(k -> newValue.onNext(
                 buildSubscribeMessage(k.getQueryName(), k.getResultType(), UUID.randomUUID().toString())
         ));
         responseObserver.enableFlowControl();
-        logger.info("QueryChannel for context '{}' connected, {} registrations resubscribed", context, queryHandlers.size());
+        if (outboundQueryStream.get() == newValue) {
+            logger.info("QueryChannel for context '{}' connected, {} registrations resubscribed", context, queryHandlers.size());
```

Comment:
Nit: This is not per definition a `resubscribed` right? It can also be the first connection, making it a `subscribed`.